File: borsa.cc
#include <iostream>

using namespace std;

int main(void)
{
	int a[] = {13,24,7,8,6,15,2,21,17,7,3,19,20,1};
	int minore, diffp, diff = 0, x, y;
	int tot = sizeof a/sizeof(int);
	
	for(int i = 0; i < tot; i++)
	{
		minore = a[i];

		for(int j = i; j < tot-1; j++)
		{
			if(a[j] == minore) continue;
			else if(a[j] > minore) diffp = a[j] - minore;
			if(diffp > diff && diffp > 0) {
				x = i;
				y = j;
				diff = diffp;
			}
		}
	}
	
	cout << "Ti conviene comprare a " << a[x] << " giorno " << x+1 << " e vendere a " << a[y] << " giorno " << y+1 << endl;
	cout << "Il tuo guadagno sarebbe: " << diff;
	
	
	return 0;
}

File: python/borsa.py
class ValoreNegativo(ValueError):
    def __str__(self):
        return 'Il valore non e\' maggiore di 0'

a = []
i = 0

while i < 14:
    try:
        num = input()
        if num < 0:
            raise ValoreNegativo

        a.append(num)
        i += 1
    except ValoreNegativo, e:
        print(e)


diff = i = 0
tot = len(a)

for i in range(tot):
    minore = a[i]

    for j in range(i, tot-1):
        if a[j] == minore:
            continue
        elif a[j] > minore:
            diffp = a[j] - minore

        if diffp > diff and diffp > 0:
            x = i
            y = j
            diff = diffp


print('Ti conviene comprare a {} giorno {} e vendere a {} giorno {}'.format(a[x], x + 1, a[y], y + 1))
print('Guadagneresti ' + str(diff))